lib/absinthe/phase/error.ex

```diff
@@ -15,18 +16,21 @@ defmodule Absinthe.Phase.Error do
     message: String.t,
     phase: module,
     locations: [loc_t],
+    extra: Keyword.t
   }
 
   @doc """
   Generate a phase error that relates to a specific point in the
   document.
   """
   @spec new(Absinthe.Phase.t, String.t, loc_t | [loc_t]) :: t
-  def new(phase, message, location) do
+  @spec new(Absinthe.Phase.t, String.t, loc_t | [loc_t], Keyword.t) :: t
+  def new(phase, message, location, extra \\ []) do
     %__MODULE__{
       phase: phase,
       message: message,
-      locations: List.wrap(location)
+      locations: List.wrap(location),
+      extra: Enum.filter(extra, &filter_extra_item/1)
```

Comment:
BTW, RE filtering: if the intention is to support being able to pass on special objects that later get transformed into better error messages (or other JSON-compatible error details), that's fine -- but the filtering out should be done in whatever phase does that transformation, not left to a generic "clean the errors" part later.